cub/cub/detail/type_traits.cuh

```diff
@@ -137,17 +138,21 @@ using has_size_t = typename has_size<T>::type;
 template <typename T, typename = void>
```

Comment:
I am strongly considering just deprecating this and replacing it with `cuda::std::tuple_size`

Reply:
sorry, I'm not following. I don't understand the relation between `StaticSize` and `tuple_size`

Reply:
This is an internal type and most of the common types here already have a `tuple_size` So what types do we support that cannot get a `tuple_size`

Reply:
We discussed this and came to the conclusion that we do not want to move to tuple_size.

@gevtushenko pointed out we do not want to expose this to user provided types